### PR TITLE
[Pre-release] Bug fix for net income adjustment for no revenue/expense lines

### PR DESCRIPTION
**Pull request by fivetran-avinash**
Test branch for the following changes. This is a pre-release. 

## Bug Fix
- Updated the logic in `int_quickbooks__retained_earnings` to ensure accounting periods with no revenue and expense class lines were accounted for. 
  - This will ensure the net income adjustment is available regardless of existing revenue or expenses.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# dbt_quickbooks v0.17.0
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:1]: Aren't we planning on cutting this as a pre-release first? If so, let's document this as v0.17.0-a1. We will then need to make the same update in the README (we can leave the dbt_project.ymls as they are)

File: README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ Include the following QuickBooks package version in your `packages.yml` file.
 ```yaml
 packages:
   - package: fivetran/quickbooks
-    version: [">=0.16.0", "<0.17.0"] # we recommend using ranges to capture non-breaking changes automatically
+    version: [">=0.17.0", "<0.18.0"] # we recommend using ranges to capture non-breaking changes automatically
```
Comment (fivetran-joemarkiewicz) [inline README.md:76]: If this will be a pre-release we should document this as a single pre-release version and not a range.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
This looks good, but are we planning for this to be a pre-release or an official release? If pre-release please make the two changes requested below.

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-avinash) [reply CHANGELOG.md]: updated

File: README.md (hunk 2 above)
Comment (fivetran-avinash) [reply README.md]: Fixed

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz Thanks for the quick checks! This PR is ready for re-review.

File: README.md [hunk 3]
```diff
@@ -73,7 +73,7 @@ Include the following QuickBooks package version in your `packages.yml` file.
 ```yaml
 packages:
   - package: fivetran/quickbooks
-    version: [">=0.16.0", "<0.17.0"] # we recommend using ranges to capture non-breaking changes automatically
+    version: v0.17.0-a1 # we recommend using ranges to capture non-breaking changes automatically
```
Comment (fivetran-joemarkiewicz) [inline README.md:76]: The official version will not include the `v` once it's on the dbt hub.
```suggestion
    version: 0.17.0-a1 # we recommend using ranges to capture non-breaking changes automatically
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM with the one small adjustment in the README. This will be fine to start the release process for the pre-release tomorrow.

**Review by fivetran-catfritz — APPROVED** [review]
lgtm

**Pull request merged**